cli/src/pcluster/cli/commands/pcluster3_configuration_converter.py

```diff
@@ -0,0 +1,593 @@
+            self.error("Error parsing configuration file {0}.\n{1}".format(self.config_file, str(e)))
+            raise
+
+    def validate(self):
+        self.validate_cluster_section_name()
+        self.get_vpc_settings()
+        self.validate_required_fields()
+        self.validate_deprecated_fields()
+
+    def convert_to_pcluster3_config(self):
+        conver_function_mappings = {
+            "Region": self.convert_region,
+            "Image": self.convert_image,
+            "Iam": self.convert_iam,
+            "AdditionalPackage": self.convert_addtional_packages,
+            "Tags": self.convert_tags,
+            "SharedStorage": self.convert_shared_storage,
+            "Monitoring": self.convert_monitoring,
+            "HeadNode": self.convert_headnode,
+            "Scheduling": self.convert_scheduling,
+        }
+        for section, convert_function in conver_function_mappings.items():
+            convert_function(section)
+
+    def convert_region(self, section):
+        self.convert_single_field("aws", "aws_region_name", self.pcluster3_configuration, section)
+
+    def convert_image(self, section):
+        image = {}
+        self.convert_single_field(self.cluster_section_name, "custom_ami", image, "CustomAmi")
+        self.convert_single_field(self.cluster_section_name, "base_os", image, "Os")
+        if image:
+            self.pcluster3_configuration[section] = image
+
+    def convert_iam(self, section):
+        iam = {}
+        role = {}
+        iam_lambda_role = self.config_parser.get(self.cluster_section_name, "iam_lambda_role", fallback=None)
+        if iam_lambda_role:
+            role["LambdaFunctionRole"] = _role_name_to_arn(iam_lambda_role)
+        if role:
+            iam["Role"] = role
+        if iam:
+            self.pcluster3_configuration[section] = iam
+
+    def convert_addtional_packages(self, section):
+        additional_packages = {}
+        intel_software = {}
+        self.convert_single_field(
+            self.cluster_section_name, "enable_intel_hpc_platform", intel_software, "IntelHpcPlatform", "getboolean"
+        )
+        if intel_software:
+            additional_packages["IntelSoftware"] = intel_software
+        if additional_packages:
+            self.pcluster3_configuration[section] = additional_packages
+
+    def convert_tags(self, section):
+        tags = self.config_parser.get(self.cluster_section_name, "tags", fallback=None)
+        if tags:
+            try:
+                tags = json.loads(tags)
+            except Exception as e:
+                self.error(f"Tags should be in the format of dictionary, {e}")
+            self.pcluster3_configuration["Tags"] = []
+            for key, value in tags.items():
+                self.pcluster3_configuration[section].append({"Key": key, "Value": value})
+
+    def convert_shared_storage(self, section):
+        shared_storage = []
+        function_section_mappings = {
```

Comment:
Feels more intuitive to map the string to the function here, even though they are just looped over as pairs below

Reply:
The reason we write it as {function: "String"} instead of {"String": function} is that "EbsSettings" map both self.convert_ebs_settings and self.convert_raid_settings